tests/components/onewire/test_entity_sysbus.py

```diff
@@ -0,0 +1,124 @@
+"""Tests for 1-Wire temperature sensor (device family 10, 22, 28, 3B, 42) connected on SysBus."""
+from datetime import datetime, timedelta
+from unittest.mock import PropertyMock, patch
+
+from pi1wire import (
+    InvalidCRCException,
+    NotFoundSensorException,
+    UnsupportResponseException,
+)
+
+from homeassistant.components.onewire.const import DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.setup import async_setup_component
+from homeassistant.util import slugify
+
+from tests.common import async_fire_time_changed, mock_registry
+
+MOCK_DEVICE_ID = "28-111111111111"
+MOCK_DEVICE_NAME = "My DS18B20"
+
+
+async def test_onewiredirect_setup_valid_device(hass):
+    """Test that sysbus config entry works correctly."""
+    entity_registry = mock_registry(hass)
+    config = {
+        "sensor": {
+            "platform": DOMAIN,
+            "names": {
+                MOCK_DEVICE_ID: MOCK_DEVICE_NAME,
+            },
+        }
+    }
+
+    with patch(
+        "homeassistant.components.onewire.sensor.Pi1Wire"
+    ) as mock_pi1wire, patch("pi1wire.OneWire") as mock_owsensor:
+        type(mock_owsensor).mac_address = PropertyMock(
+            return_value=MOCK_DEVICE_ID.replace("-", "")
+        )
+        mock_owsensor.get_temperature.side_effect = [
+            25.123,
+            FileNotFoundError,
+            25.223,
+            InvalidCRCException,
+            25.323,
+            NotFoundSensorException,
+            25.423,
+            UnsupportResponseException,
+            25.523,
+        ]
+        mock_pi1wire.return_value.find_all_sensors.return_value = [mock_owsensor]
+        assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+        await hass.async_block_till_done()
+
+        assert len(entity_registry.entities) == 1
+        sensor_id = "sensor." + slugify(MOCK_DEVICE_NAME) + "_temperature"
```

Comment:
We know what the entity_id will be. We don't need to use slugify. Define the entity id as a constant.